### PR TITLE
Add docs section explaining the volumeClaimDeletePolicy attribute

### PR DESCRIPTION
**Pull request by pebrc**
Fixes https://github.com/elastic/cloud-on-k8s/issues/4287

### CODE REVIEW

File: docs/orchestrating-elastic-stack-applications/elasticsearch/volume-claim-templates.asciidoc [hunk 1]
```diff
@@ -31,7 +31,27 @@ spec:
         storageClassName: standard
 ----
 
-ECK automatically deletes PersistentVolumeClaim resources if they are not required for any Elasticsearch node. The corresponding PersistentVolume may be preserved, depending on the configured link:https://kubernetes.io/docs/concepts/storage/storage-classes/#reclaim-policy[storage class reclaim policy].
+== Controlling volume claim deletion
+
+ECK automatically deletes PersistentVolumeClaim resources if the owning Elasticsearch nodes are scaled down. The corresponding PersistentVolume may be preserved, depending on the configured link:https://kubernetes.io/docs/concepts/storage/storage-classes/#reclaim-policy[storage class reclaim policy].
+
+In addition,  you can control what ECK should do with the PersistentVolumeClaims if you delete the Elasticsearch cluster alltogether through the `volumeClaimDeletePolicy` attribute.
```
Comment (thbkrkr) [inline docs/orchestrating-elastic-stack-applications/elasticsearch/volume-claim-templates.asciidoc:38]: Double spaces:
```suggestion
In addition, you can control what ECK should do with the PersistentVolumeClaims if you delete the Elasticsearch cluster alltogether through the `volumeClaimDeletePolicy` attribute.
```

File: docs/orchestrating-elastic-stack-applications/elasticsearch/volume-claim-templates.asciidoc [hunk 2]
```diff
@@ -31,7 +31,27 @@ spec:
         storageClassName: standard
 ----
 
-ECK automatically deletes PersistentVolumeClaim resources if they are not required for any Elasticsearch node. The corresponding PersistentVolume may be preserved, depending on the configured link:https://kubernetes.io/docs/concepts/storage/storage-classes/#reclaim-policy[storage class reclaim policy].
+== Controlling volume claim deletion
+
+ECK automatically deletes PersistentVolumeClaim resources if the owning Elasticsearch nodes are scaled down. The corresponding PersistentVolume may be preserved, depending on the configured link:https://kubernetes.io/docs/concepts/storage/storage-classes/#reclaim-policy[storage class reclaim policy].
```
Comment (thbkrkr) [inline docs/orchestrating-elastic-stack-applications/elasticsearch/volume-claim-templates.asciidoc:36]: Plural?
```suggestion
ECK automatically deletes PersistentVolumeClaim resources if the owning Elasticsearch nodes are scaled down. The corresponding PersistentVolumes may be preserved, depending on the configured link:https://kubernetes.io/docs/concepts/storage/storage-classes/#reclaim-policy[storage class reclaim policy].
```

File: docs/orchestrating-elastic-stack-applications/elasticsearch/volume-claim-templates.asciidoc [hunk 3]
```diff
@@ -31,7 +31,27 @@ spec:
         storageClassName: standard
 ----
 
-ECK automatically deletes PersistentVolumeClaim resources if they are not required for any Elasticsearch node. The corresponding PersistentVolume may be preserved, depending on the configured link:https://kubernetes.io/docs/concepts/storage/storage-classes/#reclaim-policy[storage class reclaim policy].
+== Controlling volume claim deletion
```
Comment (thbkrkr) [inline docs/orchestrating-elastic-stack-applications/elasticsearch/volume-claim-templates.asciidoc:34]:
```suggestion
== Controlling persistent volume claim deletion
```

File: docs/orchestrating-elastic-stack-applications/elasticsearch/volume-claim-templates.asciidoc (hunk 1 above)
Comment (alaudazzi) [inline docs/orchestrating-elastic-stack-applications/elasticsearch/volume-claim-templates.asciidoc:38]:
```suggestion
In addition,  you can control what ECK should do with the PersistentVolumeClaims if you delete the Elasticsearch cluster altogether through the `volumeClaimDeletePolicy` attribute.
```

File: docs/orchestrating-elastic-stack-applications/elasticsearch/volume-claim-templates.asciidoc [hunk 4]
```diff
@@ -31,7 +31,27 @@ spec:
...
+  name: es
+spec:
+  version: {version}
+  volumeClaimDeletePolicy: DeleteOnScaledownOnly
+  nodeSets:
+  - name: default
+    count: 3
+----
+
+The possible values are `DeleteOnScaledownAndClusterDeletion` and `DeleteOnScaledownOnly`. By default `DeleteOnScaledownAndClusterDeletion` is in effect which means that all PersistentVolumeClaims are deleted together with the Elasticsearch cluster. `DeleteOnScaledownOnly` however keeps the PersistentVolumeClaims when deleting the Elasticsearch cluster. If you recreate a deleted cluster with the same name and node sets as before, the existing PersistentVolumeClaims will be adopted by the new cluster.
```
Comment (alaudazzi) [inline docs/orchestrating-elastic-stack-applications/elasticsearch/volume-claim-templates.asciidoc:54]:
```suggestion
The possible values are `DeleteOnScaledownAndClusterDeletion` and `DeleteOnScaledownOnly`. By default `DeleteOnScaledownAndClusterDeletion` is in effect, which means that all PersistentVolumeClaims are deleted together with the Elasticsearch cluster. However, `DeleteOnScaledownOnly` keeps the PersistentVolumeClaims when deleting the Elasticsearch cluster. If you recreate a deleted cluster with the same name and node sets as before, the existing PersistentVolumeClaims will be adopted by the new cluster.
```

**Review by alaudazzi — APPROVED** [review]
Minor editing suggestions. LGTM.

File: docs/orchestrating-elastic-stack-applications/elasticsearch/volume-claim-templates.asciidoc (hunk 3 above)
Comment (pebrc) [reply docs/orchestrating-elastic-stack-applications/elasticsearch/volume-claim-templates.asciidoc:34]: I think I leave it as is as we are consistently just speaking of "volume claims" in the titles of this chapter. Persistent is implied.

**Review by thbkrkr — APPROVED** [review]
LGTM

**Pull request merged**